osc-server/src/main/java/org/osc/core/broker/service/securitygroup/BindSecurityGroupService.java

```diff
@@ -54,275 +56,277 @@
+					}
+				}
+
+				FailurePolicyType failurePolicyType = this.apiFactoryService.supportsFailurePolicy(this.securityGroup)
+						? serviceToBindTo.getFailurePolicyType() : FailurePolicyType.NA;
+
+				SecurityGroupInterface sgi = SecurityGroupInterfaceEntityMgr
+						.findSecurityGroupInterfacesByVsAndSecurityGroup(em, vs, this.securityGroup);
+				if (sgi == null) {
+					// If the policy is null the tag should also be null
+					Long tag = policy == null ? null : VirtualSystemEntityMgr.generateUniqueTag(em, vs);
+					String tagString = tag == null ? null : SecurityGroupInterface.ISC_TAG_PREFIX + tag;
+
+					// Create a new security group interface for this service
+					sgi = new SecurityGroupInterface(vs, policies, tagString,
+							org.osc.core.broker.model.entities.virtualization.FailurePolicyType
+									.valueOf(failurePolicyType.name()),
+							order);
+
+					SecurityGroupInterfaceEntityMgr.toEntity(sgi, this.securityGroup, serviceToBindTo.getName());
+
+					log.info("Creating security group interface " + sgi.getName());
+					OSCEntityManager.create(em, sgi, this.txBroadcastUtil);
+
+					this.securityGroup.addSecurityGroupInterface(sgi);
+					sgi.setSecurityGroup(this.securityGroup);
+					OSCEntityManager.update(em, this.securityGroup, this.txBroadcastUtil);
+				} else {
+					if (hasServiceChanged(sgi, serviceToBindTo, policies, order)) {
+						log.info("Updating Security group interface " + sgi.getName());
+						sgi.setPolicies(policies);
+						sgi.setFailurePolicyType(org.osc.core.broker.model.entities.virtualization.FailurePolicyType
+								.valueOf(failurePolicyType.name()));
+						sgi.setOrder(order);
+					}
+					OSCEntityManager.update(em, sgi, this.txBroadcastUtil);
+				}
+
+				order++;
+			}
+
+			// Go through all the security group interfaces and if they are not selected, remove them
+			for (SecurityGroupInterface sgi : this.securityGroup.getSecurityGroupInterfaces()) {
+				boolean isServiceSelected = isServiceSelected(servicesToBindTo, sgi.getVirtualSystem().getId());
+				if (!isServiceSelected || sgi.getMarkedForDeletion()) {
+					log.info("Marking service " + sgi.getName() + " for deletion");
+					OSCEntityManager.markDeleted(em, sgi, this.txBroadcastUtil);
+				}
+			}
+
+			this.txBroadcastUtil.addMessageToMap(this.securityGroup.getId(),
+					this.securityGroup.getClass().getSimpleName(), EventType.UPDATED);
+
+			Job job = this.conformService.startBindSecurityGroupConformanceJob(em, this.securityGroup, unlockTask);
+
+			response = new BaseJobResponse(job.getId());
+
+		} catch (Exception e) {
+			LockUtil.releaseLocks(unlockTask);
+			throw e;
+		}
+
+		return response;
+	}
+
+	private boolean hasServiceChanged(SecurityGroupInterface sgi, VirtualSystemPolicyBindingDto serviceToBindTo,
+			Set<Policy> policies, long order) {
+		boolean policyChanged = policies != null && (sgi.getPolicies() == null || !sgi.getPolicies().equals(policies));
```

Comment:
we cannot use a equals comparision for a list. Iterate through the list to make sure they are really equal by comparing the policy objects